Chrysalis-Papilio/Soen490-Papilio-Website
# Papilio 84 task - Address autofill integration

## General info

To increase the UX in the form for the business creation, we implemented autofill for the address of the company. It takes an input in the first line of the address (the street name and house number) and search through through it's database an address with all the information (postal code, country, city, province/state, ...) and fills the missing information in their corresponding field.

Since the address is now saved inside a single field in the database, the address is then combine together into a single field in the backend of the frontend before sending it to the backend. 

_Issue number_: Papilio-84

Task description:

- Autofill address in business form
- combine address information into a single field
- test

## Manual Test

To test the address input you have to create your businessId then visit the profile form. Then you can submit the business and the business should be created.

__Make sure that you have the appropriate variables in the backend .env file. (Contact me if you don't)__
__You need a Token to use mapbox autofill. (contact me to get it)__

1. start the backend
```npm run build && npm start```
2. start the website
```npm start```
3. on the main page add the businessId then click on create business
4. Add information for the business, including the address
5. Submit the business form

##Testing

To run all the unit tests, run the following code: ```npm test```
To make sure that the tests have reach the appropriate level of coverage, run the following code: ```npm run coverage```

## Additional Notes

- In the current state of this branch, you won't be able to submit the form because you are missing the email of the business. This is a requirement in the backend now. However, this has been added in #19 . Once, the 2 are merge, it will work correctly.

- I wasn't sure of the request message anymore. I left the address object inside the request object. I think I can delete it but I was sure so I left it there.
```
{
    business: ...,
    address: ...,
    employee: ...
}
```
